### DISCUSSION TITLE
The agent has no identities.

### DISCUSSION THREAD

**Discussion by Zomnipotential** [community/community#50701] [New to GitHub]
### Select Topic Area

Question

### Body

Dear all, Hi,
***
Running on macos Ventura 13.2.1
Using macos Terminal
***
Totally new to github, been trying for 3 hours now to follow the github's own instructions step by step, to generate GPG and SSH keys on my computer and save them in GitHub. Have already created a repository. "git remote -v" generates "git@github.com:Zomnipotential/repo (fetch)" and dito for "push". But when I try "git push origin main" from inside the "venv" it fails with the error:
"git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists."
What should I do? I have upcoming deadlines in a couple hours and still no solution. Have been searching the web with no result.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Zomnipotential ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Reply by Zomnipotential** [reply]
Thanx a lot; I'll try to remember that.

**Reply by Zomnipotential** [reply]
I really hope someone will take time and answer this question. Annoying that I have generated the SSH key 3 times from scratch and it still doesn't work

**Comment by airtower-luna** [comment]
What's the output of `ssh -v git@github.com`? That should give you debug info on the SSH connection itself, including keys it's using (or trying to).

**Reply by Zomnipotential** [reply]
I solved it but thank you anyways. I think I had generated the key from inside a virtual environment but then I tried to connect from outside. I don't remember the details (short memory fatal error :-)) but I think that's the reason.
